### PR TITLE
feat(patterns): new tables documentation

### PR DESCRIPTION
**Pull request by hzhu**
## Description

This pull request introduces a new "Patterns" section with the first set of patterns (Tables patterns).

## Checklist

- [x] :ok_hand: website updates are Garden Designer approved (add the designer as a reviewer)
- [x] :black_nib: copy updates are approved (add the content strategist as a reviewer)
- [x] :link: considered opportunities for adding cross-reference URLs (grep for keywords)
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

**Comment by jzempel** [comment]
@hzhu can you break the site-wide style changes into a separate PR? Those items are a separate unit of work that should not impede the table pattern doc and warrant a discussion of their own – even if the discussion is about aligning team perspective as it pertains to Garden component usage and the website.

**Comment by hzhu** [comment]
@jzempel sounds good, I created https://github.com/zendeskgarden/website/pull/343 dedicated to the site-wide typography spacing changes.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (diana-oum) [inline src/pages/patterns/tables/basic-formatting.mdx:37]: Should say Header cells categorize and organize the data within the table. "

So information --> data

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (diana-oum) [inline src/pages/patterns/tables/basic-formatting.mdx:127]: Accidental capital A for align. Should be "Left align" like the others

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (diana-oum) [inline src/pages/patterns/tables/basic-formatting.mdx:256]: This reads funny. I think we left out "is"? "If any ordering is applied to a table including default ordering, the sorting icon should be present." I'll update the docs too cause they were missing this.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (diana-oum) [inline src/pages/patterns/tables/basic-formatting.mdx:257]: @james-sann Reading this sentence again: "This indicates how the table is being ordered and also creates a better affordance for the control."

What if we changed it to "This indicates how the table is being ordered and also creates a clear affordance to control."?

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (diana-oum) [inline src/pages/patterns/tables/basic-formatting.mdx:299]: Lowercase a in External anchor since that's how it's presented in Garden today. Updated the docs to reflect this too.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (diana-oum) [inline src/pages/patterns/tables/basic-formatting.mdx:282]: This is formatted as a bullet in the google doc. @james-sann - Did we change it to a sentence? If so, it's missing a period

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (diana-oum) [inline src/pages/patterns/tables/basic-formatting.mdx:417]: No need for period after feedback since it's a link :)

**Review by diana-oum — CHANGES_REQUESTED** [review]
Small content tweaks -- some of them I want to double check with James and let him weigh in before going with the change.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (hzhu) [reply src/pages/patterns/tables/basic-formatting.mdx:282]: It looks like this is also formatted as a bullet in Figma. The mistake is on the code and I'll get this fixed 👍🏽 .

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (jzempel) [inline src/pages/patterns/tables/basic-formatting.mdx:356]: Just want to double-check that we want to call out the "60%" unit. There have been previous concerns re: Garden's `Skeleton` implementation being too slim and not accurately simulating the rendered text. If we address that concern, we'll have to keep this text up-to-date.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (jzempel) [inline src/pages/patterns/tables/basic-formatting.mdx:402]: "Screen reader order"? (capitalization)

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (jzempel) [inline src/pages/patterns/tables/basic-formatting.mdx:405]: Should we index on the specifics of screen reader technology? Or should our pattern docs make claims regarding assistive technology in general? Might be a question for the a11y folks as we continue forward with pattern docs.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (jzempel) [inline src/pages/patterns/tables/basic-formatting.mdx:415]: @hzhu please connect with me offline re: the form before this PR is merged. I need to do a permissions audit.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (jzempel) [inline src/pages/patterns/tables/basic-formatting.mdx:394]: There should be a general-use `Figure` component built for these that ties to Garden's theming ... and no use of `<br>`. See [BestPractice](https://github.com/zendeskgarden/website/blob/main/src/components/MarkdownProvider/components/BestPractice.tsx) for reference.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (hzhu) [reply src/pages/patterns/tables/basic-formatting.mdx:282]: Updated and added backticking for the html attribute 👍🏽

File: src/pages/patterns/examples/tables/TableCursorPagination.tsx
Comment (Francois-Esquire) [inline src/pages/patterns/examples/tables/TableCursorPagination.tsx:25]: Would it be better to offset the index by +1 and start the count at 1 instead of 0 for demo purposes? Starting a count at 0 feels very "engineering" to me.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx:135]: @diana-oum Agreed. Updated the WIP and Ready to Ship Google docs to reflect this.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx:256]: @diana-oum Thank you. Also updated the Ready to Build latest Figma to reflect this.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx:257]: @diana-oum I like the tone better. I'll update the Google Docs and Figma to reflect this change.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx:305]: Reflects the [existing Garden content design guidelines for date and time](https://garden.zendesk.com/content/grammar#date-and-time)

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx:356]: @jzempel I'm not familiar with that concern. Previous pattern documentation by Allison pointed to the 60% and it wasn't raised in any prior design reviews of this content. Happy to update this whether that be in this release or an iteration in the future based on the data at hand.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx:405]: @jzempel Good point! This documentation was shown to the accessibility team (Jen) without any previous concerns. Happy to iterate in the future however.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (diana-oum) [reply src/pages/patterns/tables/basic-formatting.mdx:32]: Yes since that's the capitalization rule that exists today and we still need to discuss as a team on how we want to move forward in changing this.

cc: @james-sann I've updated our docs

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (diana-oum) [reply src/pages/patterns/tables/basic-formatting.mdx:402]: Yes it should be "Screen reader order" as it is in the Google doc. Might have been a Google doc --> Figma transfer issue.

File: src/layouts/Home/components/Patterns.tsx
Comment (diana-oum) [inline src/layouts/Home/components/Patterns.tsx:72]: @hzhu @james-sann - I'm curious why this changed? Also just fyi I wasn't looped in on this content in the beginning so would love any additional context. Is this how the team is defining a pattern?

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (hzhu) [reply src/pages/patterns/tables/basic-formatting.mdx:216]: I think this is prettier formatting to adjust for max character per line rule.

File: src/layouts/Home/components/Patterns.tsx
Comment (hzhu) [reply src/layouts/Home/components/Patterns.tsx:72]: This came from the [buttons draft PR](https://github.com/zendeskgarden/website/pull/260/files#diff-fef5f8c3fdd869fc68a64da111007a8560b21a0a0db0ab106b9d21dd5dbc884fR72). @jzempel do you have context on this copy?

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (diana-oum) [reply src/pages/patterns/tables/basic-formatting.mdx:46]: No, didn't catch this one in the docs. Let's change this to "text" and it will work just fine.

cc: @james-sann I'm updating the docs.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (diana-oum) [reply src/pages/patterns/tables/basic-formatting.mdx:338]: Yes, updating docs.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (diana-oum) [reply src/pages/patterns/tables/basic-formatting.mdx]: Let's keep it consistent as the previous line. I'll update the docs to reflect this.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (diana-oum) [reply src/pages/patterns/tables/basic-formatting.mdx:305]: Hm, it does look strange in the image. Can we change that to an actual example like the other entries instead of exposing the formula? I'm concerned other folks will also be confused. Since the content guidelines are linked, the formula is there for folks to refer back to.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (mtomcal) [inline src/pages/patterns/tables/basic-formatting.mdx:267]: I am trying to review this pattern holistically from an accessibility standpoint here:

Is the accessibility support from react-beautiful-dnd enough for documenting how to do accessibility for Garden Table row reordering? I ran through the screen reader and it said "you have moved item from position 1 to position 2". Any chance there's a way to recommend how to change "item" to "Raspberry" or an identifying row name for developers to follow? What do you think @jgorfine-zendesk?

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (mtomcal) [inline src/pages/patterns/tables/basic-formatting.mdx:277]: Should we also have a recommendation on column reordering as I have seen that topic come up quite frequently?

File: src/layouts/Home/components/Patterns.tsx
Comment (jzempel) [reply src/layouts/Home/components/Patterns.tsx:72]: The #260 draft PR that Henry mentions was the original PR for adding the "Patterns" pillar to the website – and the baseline for this PR. This tagline was set at Content Design's request.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx:277]: @mtomcal Until that interaction pattern is been appropriately reviewed by Garden and the Accessibility team and documented, we can start with a recommendation.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx:49]: Confirmed and updated all the docs to represent this.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx]: @diana-oum Thank you!

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx]: @diana-oum @hzhu Henry has changed this to 'text length' in the build. @hzhu Let me know if you want to alter this to just "text" so I can update the design and Google docs to reflect this

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx:71]: @hzhu If you have scope to change this now for the first release, happy with this suggestion. Otherwise, we can change in the next release.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx:91]: Good point. I see the pattern guidelines being users who can actively utilize the Garden components.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx:141]: Thanks for the technical consideration. I still feel like the design direction is still to have icons or avatars center aligned, especially if there's table header text in the same column that exceeds the width of the icon or avatar.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx:161]: Good catch. Will update docs to reflect this ccing @hzhu

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx:216]: Happy to update this. @hzhu Updated Figma design to reflect this

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx:305]: @diana-oum Good point. I'll update the screenshots ccing @hzhu

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (jgorfine-zendesk) [reply src/pages/patterns/tables/basic-formatting.mdx:267]: Hey @mtomcal — I totally agree with you: it might be more helpful if the label were more specific, i.e., had the name of the "item" in it. 

I wonder if it would also be helpful to include the total number of positions. Looking at the example from the React Beautiful DnD guidelines:
```
You have moved the task to position 2
``` 

it could be modified to be like this: 
```
You have moved the task “Raspberry“ to position 2 of 5
```

In the context of a table, that might sound like this:
```
You have moved [string value of row's <th> element] to row 2 of 5
```

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (hzhu) [reply src/pages/patterns/tables/basic-formatting.mdx:71]: Sounds good, I created tasks for this in the backlog for a follow-on release.

**Comment by hzhu** [comment]
@jzempel 

Let me know if the new file structure works. The three duplicate files have been moved into a `shared` directory and all example files now live in `examples` folder at the `src` level.

File: src/pages/patterns/examples/tables/TableCursorPagination.tsx
Comment (hzhu) [reply src/pages/patterns/examples/tables/TableCursorPagination.tsx]: Well, it is a code example 😛  and the examples come from the production component [page](https://garden.zendesk.com/components/table#pagination). 0 based indices seem okay to me, but feel free to open a new task to change that.

File: src/pages/patterns/examples/tables/TableSize.tsx
Comment (hzhu) [reply src/pages/patterns/examples/tables/TableSize.tsx]: If I recall, the standard in `react-components` was to avoid inline styles and prefer styled components? I'll leave that update to a different PR since I've refactored the example files and this file is no longer modified.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (hzhu) [reply src/pages/patterns/tables/basic-formatting.mdx:267]: Could these potential changes be reserved for a separate unit of work since this example currently exists [in production](https://garden.zendesk.com/components/table#draggable)? I'm happy to create a task for it if it's necessary.

File: src/pages/patterns/examples/tables/TableVerticalScrolling.tsx
Comment (hzhu) [reply src/pages/patterns/examples/tables/TableVerticalScrolling.tsx]: Seems like removing the px unit breaks the scroll example.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (hzhu) [reply src/pages/patterns/tables/basic-formatting.mdx:216]: I see, I thought the original comment referred to the new line break on line 216. In any case, I have updated this to match the Figma which combines this sentence and the previous into one paragraph.

File: src/layouts/Home/components/Patterns.tsx
Comment (hzhu) [reply src/layouts/Home/components/Patterns.tsx]: @diana-oum thanks for looking into it and providing the reasons. The latest changes have been pushed up. Feel free to re-open this comment for further discussion.

**Review by james-sann — APPROVED** [review]
All the updates look fine to me for v1. Thanks Henry for all your effort and iteration here 🙌

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (mtomcal) [reply src/pages/patterns/tables/basic-formatting.mdx:269]: Definitely task for this as a tech debt item for a11y since this predates this effort. 👍

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (diana-oum) [reply src/pages/patterns/tables/basic-formatting.mdx:49]: @hzhu ellipsis needs to be linked and capitalized

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (diana-oum) [inline src/pages/patterns/tables/basic-formatting.mdx:303]: @hzhu Time should be lowercase
@james-sann Figma has it capitalized. Let's be careful with the transfer from Google docs for future Garden documentation!

**Review by diana-oum — CHANGES_REQUESTED** [review]
Thanks for fixes! @hzhu Once these are done, I think it's ready to go.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (jzempel) [inline src/pages/patterns/tables/basic-formatting.mdx:403]: We'll either need to remove and do a (quick) follow-on to add this back, or wait for my doc review.

**Review by jzempel — CHANGES_REQUESTED** [review]
I'm not sure I see the value of injecting a `shared/` directory in examples. Just feels like one more thing to maintain. I would recommend keeping those files where they were. The team should have expectations that an example could be used in multiple locations.

File: src/pages/components/table.mdx
Comment (jzempel) [inline src/pages/components/table.mdx:45]: per previous review comment, these moves should not be necessary.

**Review by jzempel — CHANGES_REQUESTED** [review]

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (smritimadan) [reply src/pages/patterns/tables/basic-formatting.mdx]: Trying to understand how much effort this will be exactly @hzhu / @jzempel. 

If it is more effort than initially anticipated, could we break it out into a separate fast-follow ticket? Any concerns with that @james-sann / @nyckie ?

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx:164]: @hzhu "When creating additional categorization of similar items, use grouped rows." Grouped rows to be linked to the component. See Figma docs and comments.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx:39]: @diana-oum Hmmmmm I don't think these were ever numbered in the Figma design files (maybe you looked at the screenshots from Henry's build within the Figma file?).

@hzhu This should be an unordered (bulleted) list.

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx:303]: @diana-oum Good catch!

@hzhu Fixed in Figma

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx:319]: @diana-oum Okay!

@hzhu Figma file updated to reflect this

File: src/pages/patterns/tables/basic-formatting.mdx
Comment (james-sann) [reply src/pages/patterns/tables/basic-formatting.mdx:259]: ccing @hzhu

**Review by jzempel — APPROVED** [review]

**Pull request merged**
